frame/curve-amm/src/lib.rs

```diff
@@ -601,172 +673,114 @@ pub mod pallet {
 		fn exchange(
 			who: &Self::AccountId,
 			pool_id: T::PoolId,
-			i_token: T::PoolTokenIndex,
-			j_token: T::PoolTokenIndex,
+			asset_id: Self::AssetId,
 			dx: Self::Balance,
 			min_dy: Self::Balance,
+			keep_alive: bool,
 		) -> Result<Self::Balance, DispatchError> {
-			let prec = T::Precision::get();
 			let zero_b = Self::Balance::zero();
 			ensure!(dx >= zero_b, Error::<T>::AssetAmountMustBePositiveNumber);
-
 			let pool = Self::get_pool_info(pool_id).ok_or(Error::<T>::PoolNotFound)?;
-			let i = i_token.into() as usize;
-			let j = j_token.into() as usize;
-
 			let assets_pair = PoolAssets::<T>::get(pool_id).ok_or(Error::<T>::PoolNotFound)?;
-			let assets = assets_pair.as_slice();
-			let mut balances = Vec::new();
-			for asset_id in assets {
-				balances.push(PoolAssetBalance::<T>::get(pool_id, asset_id));
+			let mut base_asset = assets_pair.base;
+			let mut quote_asset = assets_pair.quote;
+			if asset_id == quote_asset {
+				base_asset = assets_pair.quote;
+				quote_asset = assets_pair.base;
+			} else if asset_id != base_asset {
+				return Err(Error::<T>::AssetNotFound.into())
 			}
```

Comment:
This part of code has been used a minimum of 2 times, maybe you should create a function for it?